src/test/java/com/commercetools/project/sync/SyncerFactoryTest.java

```diff
@@ -329,20 +461,86 @@ void sync_WithErrorOnFetch_ShouldCloseClientAndCompleteExceptionally() {
     when(sourceClient.execute(any(InventoryEntryQuery.class)))
         .thenReturn(CompletableFutureUtils.exceptionallyCompletedFuture(badGatewayException));
 
-    stubClientsCustomObjectService(targetClient);
+    final ZonedDateTime currentCtpTimestamp = ZonedDateTime.now();
+    stubClientsCustomObjectService(targetClient, currentCtpTimestamp);
 
-    final SyncerFactory syncerFactory = SyncerFactory.of(() -> sourceClient, () -> targetClient);
+    final SyncerFactory syncerFactory =
+        SyncerFactory.of(() -> sourceClient, () -> targetClient, getMockedClock());
 
     // test
     final CompletionStage<Void> result = syncerFactory.sync("inventoryEntries");
 
     // assertions
+    verifyTimestampGeneratorCustomObjectUpsert(targetClient, 1);
     verify(sourceClient, times(1)).execute(any(InventoryEntryQuery.class));
     verifyInteractionsWithClientAfterSync(sourceClient, 1);
     assertThat(result).hasFailedWithThrowableThat().isExactlyInstanceOf(BadGatewayException.class);
   }
 
   @Test
+  @SuppressWarnings("unchecked")
+  void
+      sync_WithErrorOnCurrentCtpTimestampUpsert_ShouldCloseClientAndCompleteExceptionallyWithoutSyncing() {
```

Comment:
> ShouldCloseClient

Could you point to the place where exactly is the check for a closed client? 🙂

Reply:
@LEQADA https://github.com/commercetools/commercetools-project-sync/pull/25/files/be61681c30d29acd2817603f637d414335ff26f9#diff-121581e46953834fbe6e01722b2941a1R504 this line calls this method:

`TestUtils#verifyInteractionsWithClientAfterSync` which asserts client closure on it's first line